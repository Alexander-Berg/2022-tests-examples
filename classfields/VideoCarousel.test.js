const React = require('react');
const { shallow } = require('enzyme');

const contextMock = require('autoru-frontend/mocks/contextMock').default;
const videoList = require('autoru-frontend/mocks/youtubeVideosMock.js');

const VideoCarousel = require('./VideoCarousel');

it('должен отрендерить блок видео для валидной марки', () => {
    const pageParams = {
        mark: 'AUDI',
    };
    const tree = shallow(
        <VideoCarousel
            className="PageCard__videoCarousel"
            params={ pageParams }
            videoList={ videoList }
            isCardPage={ true }
        />,
        { context: contextMock },
    );
    expect(tree).toMatchSnapshot();
});

const excludedVideoMarks = require('auto-core/data/video/excluded-mark-list.json');

excludedVideoMarks.forEach(mark => {
    it(`не должен отрендерить блок видео для марки ${ mark } из списка исключений`, () => {
        const pageParams = {
            mark,
        };
        const tree = shallow(
            <VideoCarousel
                className="PageCard__videoCarousel"
                params={ pageParams }
                videoList={ videoList }
                isCardPage={ true }
            />,
            { context: contextMock },
        );
        expect(tree).toMatchSnapshot();
    });
});
