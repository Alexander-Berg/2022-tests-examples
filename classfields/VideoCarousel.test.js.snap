// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`должен отрендерить блок видео для валидной марки 1`] = `
<AmpSlider
  className="VideoCarousel VideoCarousel_page_card PageCard__videoCarousel"
  footer={
    <AmpButton
      color="transparentBlue"
      size="l"
      type="link"
      url="link/video/?mark=AUDI&configuration_id="
      width="full"
    >
      Все видео
    </AmpButton>
  }
  items={
    Array [
      <div
        className="VideoCarousel__item"
      >
        <amp-lightbox
          data-amp-bind-open="showLightbox0"
          id="lightbox0"
          layout="nodisplay"
          on="lightboxClose:AMP.setState({showLightbox0: false})"
        >
          <div
            className="lightbox"
            role="button"
            tabIndex="0"
          >
            <amp-youtube
              class="lightbox__video"
              data-videoid="wLQIpkHdMns"
              height="270"
              layout="responsive"
              width="480"
            />
            <div
              className="lightbox__close"
              on="tap:lightbox0.close"
              role="button"
              tabIndex="0"
            />
          </div>
        </amp-lightbox>
        <div
          className="VideoCarousel__itemPreview"
        >
          <AmpImg
            className="VideoCarousel__itemPreviewImg"
            height="86"
            layout="responsive"
            src="https://i.ytimg.com/vi/wLQIpkHdMns/mqdefault.jpg"
            width="152"
          />
          <div
            className="VideoCarousel__itemPlay"
            on="tap:AMP.setState({ showLightbox0: true })"
            role="button"
            tabIndex="0"
          >
            <AmpIcon
              id="svgID"
              size="48"
            />
          </div>
          <div
            className="VideoCarousel__itemDuration"
          >
            22:42
          </div>
        </div>
        <div
          className="VideoCarousel__itemTitle"
        >
          2017 Audi A3 E-tron Sportback...
        </div>
      </div>,
      <div
        className="VideoCarousel__item"
      >
        <amp-lightbox
          data-amp-bind-open="showLightbox1"
          id="lightbox1"
          layout="nodisplay"
          on="lightboxClose:AMP.setState({showLightbox1: false})"
        >
          <div
            className="lightbox"
            role="button"
            tabIndex="0"
          >
            <amp-youtube
              class="lightbox__video"
              data-videoid="0T5nIu91-f4"
              height="270"
              layout="responsive"
              width="480"
            />
            <div
              className="lightbox__close"
              on="tap:lightbox1.close"
              role="button"
              tabIndex="0"
            />
          </div>
        </amp-lightbox>
        <div
          className="VideoCarousel__itemPreview"
        >
          <AmpImg
            className="VideoCarousel__itemPreviewImg"
            height="86"
            layout="responsive"
            src="https://i.ytimg.com/vi/0T5nIu91-f4/mqdefault.jpg"
            width="152"
          />
          <div
            className="VideoCarousel__itemPlay"
            on="tap:AMP.setState({ showLightbox1: true })"
            role="button"
            tabIndex="0"
          >
            <AmpIcon
              id="svgID"
              size="48"
            />
          </div>
          <div
            className="VideoCarousel__itemDuration"
          >
            06:56
          </div>
        </div>
        <div
          className="VideoCarousel__itemTitle"
        >
          AUDI A3 2020 Test Drive and Review
        </div>
      </div>,
    ]
  }
  title={
    <AmpLink
      url="link/video/?mark=AUDI&configuration_id="
    >
      Видео о модели
    </AmpLink>
  }
/>
`;

exports[`не должен отрендерить блок видео для марки AC из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ADLER из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ALPINA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ALPINE из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки AMC из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ARIEL из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ASIA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки AUSTIN из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки AVTOKAM из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BAJAJ из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BALTIJAS_DZIPS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BATMOBILE из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BEIJING из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BERTONE из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BILENKIN из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BITTER из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BORGWARD из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BRABUS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BRISTOL из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BUFORI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BUGATTI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки BYVIN из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки CALLAWAY из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки CARBODIES из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки CATERHAM из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки CHANGFENG из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки CIZETA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки COGGIOLA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки DAIMLER из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки DELAGE из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки DERWAYS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки DESOTO из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки DONINVEST из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки DS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки E_CAR из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки EAGLE из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки EAGLE_CARS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ECOMOTORS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки EXCALIBUR из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки EXCLUSIVE из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки FERRARI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки FOTON из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки FSO из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки FUQI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки GENESIS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки GEO из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки GONOW из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки GORDON из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки HANOMAG из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки HINDUSTAN из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки HOLDEN из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки HUANGHAI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки HUDSON из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки IG из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки INNOCENTI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки INVICTA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ISDERA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки JENSEN из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки JMC из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки KANONIR из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки KOENIGSEGG из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки KTM из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки LAMBORGHINI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки LIEBAO из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки LOTUS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки LTI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки MARCOS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки MARLIN из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки MASERATI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки MCLAREN из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки MEGA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки MERCURY из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки MICROCAR из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки MINELLI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки MORRIS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки NOBLE из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки OSCA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки PAGANI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки PANOZ из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки PGO из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки PIAGGIO из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки PREMIER из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки PUCH из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки PUMA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки QVALE из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки RELIANT из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки RENAISSANCE_CARS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки REZVANI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки RIMAC из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки RONART из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ROVER из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки SALEEN из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки SAMSUNG из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки SANTANA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки SHANGHAI_MAPLE из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки SPECTRE из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки SPYKER из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки STEYR из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки TALBOT из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки TATA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки TATRA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки TAZZARI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки THINK из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки TIANMA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки TIANYE из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки TOFAS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки TRAMONTANA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки TRIUMPH из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки TVR из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ULTIMA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки VAUXHALL из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки VECTOR из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки VENTURI из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки W_MOTORS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки WANDERER из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки WESTFIELD из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки WIESMANN из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ZASTAVA из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ZENOS из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ZENVO из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ZIBAR из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ZIL из списка исключений 1`] = `""`;

exports[`не должен отрендерить блок видео для марки ZX из списка исключений 1`] = `""`;
